lib/auth/profile.js

```diff
@@ -8,13 +9,33 @@ module.exports = (endpoint) => {
 
   const request = api(endpoint);
 
-  return token => {
-    const headers = {
-      Authorization: `bearer ${token}`
-    };
-    return request(`/me`, { headers })
-      .then(response => response.json.data)
-      .catch(() => null);
+  return (token, session) => {
+
+    return Promise.resolve()
+      .then(() => {
+        if (!session.profile ||
+          (session.profile &&
+            moment(new Date()).isAfter(moment(session.profile.expiresAt).format()))) {
```

Comment:
The `new Date()` argument in the calls to moment can be removed. `moment()` will default to the current date/time if no argument is passed.